plugins/ipam/static/main.go

```diff
@@ -211,12 +221,15 @@ func LoadIPAMConfig(bytes []byte, envArgs string) (*IPAMConfig, string, error) {
 	numV6 := 0
 
 	for i := range n.IPAM.Addresses {
-		ip, addr, err := net.ParseCIDR(n.IPAM.Addresses[i].AddressStr)
-		if err != nil {
-			return nil, "", fmt.Errorf("invalid CIDR %s: %s", n.IPAM.Addresses[i].AddressStr, err)
+		if n.IPAM.Addresses[i].Address.IP == nil {
```

Comment:
Should we check `n.IPAM.Address[i].Address != nil` ibefore the accesss of `n.IPAM.Addresses[i].Address.IP` for corner case? (I believe that net.ParseCIDR() returns non nil for *IPNet), however, we may need to double check...

I'm worried segmentation fault at the access of `n.IPAM.Addresses[i].Address.IP`

Reply:
> Should we check `n.IPAM.Address[i].Address != nil` ibefore the accesss of `n.IPAM.Addresses[i].Address.IP` for corner case? (I believe that net.ParseCIDR() returns non nil for *IPNet), however, we may need to double check...

I don't think `net.ParseCIDR` can return `nil` for `*IPNet` **without** also returning an error. 
https://cs.opensource.google/go/go/+/refs/tags/go1.15.13:src/net/ip.go;l=712

Furthermore, we're the ones putting the `net.Address` in the slice: https://github.com/containernetworking/plugins/pull/638/files/8ab23366fb09065838013bb464d83aaca1ac2f51#diff-0469cbf9b042cb5f9cd3a38b880957736c89e1b78770b600c1d845ca9d9be410R201
> 
> I'm worried segmentation fault at the access of `n.IPAM.Addresses[i].Address.IP`

I don't think that can happen (without `net.ParseCIDR` returning an error). But, for added safety, I can update the code to fail on

```golang
...
for _, addrStr := range n.Args.A.IPs {
	ip, addr, err := net.ParseCIDR(addrStr)
	if err != nil || addr == nil { // <==== HERE
		return nil, "", fmt.Errorf("an entry in the 'ips' field is NOT in CIDR notation, got: '%s'", addrStr)
	}
	addr.IP = ip
	n.IPAM.Addresses = append(n.IPAM.Addresses, Address{AddressStr: addrStr, Address: *addr})
}
...
```